Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

c:\Users\AD39644\Desktop\EIS\OneDrive_2024-07-25\Auditing Script\Visualizor.py:167: FutureWarning:

'M' is deprecated and will be removed in a future version, please use 'ME' instead.

Traceback (most recent call last):
  File "c:\Users\AD39644\Desktop\EIS\OneDrive_2024-07-25\Auditing Script\Visualizor.py", line 238, in <module>
    main()
  File "c:\Users\AD39644\Desktop\EIS\OneDrive_2024-07-25\Auditing Script\Visualizor.py", line 194, in main
    scheduling_html = generate_scheduling_tab()
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\AD39644\Desktop\EIS\OneDrive_2024-07-25\Auditing Script\Visualizor.py", line 168, in generate_scheduling_tab
    workdays = pd.date_range(f"{y}-{mm}-01", periods=days, freq='B')
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\AD39644\Desktop\EIS\OneDrive_2024-07-25\Auditing Script\.venv\Lib\site-packages\pandas\core\indexes\datetimes.py", line 1008, in date_range
    dtarr = DatetimeArray._generate_range(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\AD39644\Desktop\EIS\OneDrive_2024-07-25\Auditing Script\.venv\Lib\site-packages\pandas\core\arrays\datetimes.py", line 412, in _generate_range
    periods = dtl.validate_periods(periods)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\AD39644\Desktop\EIS\OneDrive_2024-07-25\Auditing Script\.venv\Lib\site-packages\pandas\core\arrays\datetimelike.py", line 2504, in validate_periods
    raise TypeError(f"periods must be a number, got {periods}")
TypeError: periods must be a number, got Index([31], dtype='int32')
